commons/zenoh-shm/src/api/cleanup.rs

```diff
@@ -0,0 +1,29 @@
+//
+// Copyright (c) 2024 ZettaScale Technology
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+//
+// Contributors:
+//   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
+//
+
+use crate::cleanup::CLEANUP;
+
+/// Make forced cleanup
+/// NOTE: this is a part of ugly on-exit-cleanup workaround and will be removed
+/// In order to properly cleanup some SHM internals upon process exit, Zenoh installs exit handlers (see atexit() API).
+/// The bad thing is that atexit handler is executed only on process exit(), the terminating signal handlers (like SIGINT)
+/// bypass it and terminate the process without cleanup. To eliminate this effect, Zenoh overrides SIGHUP, SIGTERM, SIGINT
+/// and SIGQUIT handlers and calls exit() inside to make graceful shutdown. If user is going to override these Zenoh's handlers,
+/// the workaround will break, and there are two ways to keep this workaround working:
+/// - execute overridden Zenoh handlers in overriding handler code
+/// - call forced_cleanup() anywhere at any time before terminating the process
+#[zenoh_macros::unstable_doc]
+pub fn forced_cleanup() {
```

Comment:
Well, it is not unsafe. The improper usage can break the application logic a little, but never leads to safe code having UB. Cleanup unlinks SHM segments created by exiting process from filesystem with the following consequences:
- Sessions that are not linked to this segment will fail to link it if they try. Such kind of errors are properly handled. This is where the app logic may break a little.
- Already linked processes will still have this shared memory mapped and safely accessible
- The actual memory will be reclaimed by the OS only after last process using it will close it or exit

Regarding the naming I agree

Reply:
Ok. I find the explanation in your comment more clear than what is actually written in the doc.
Ok for not marking the function as `unsafe`.